File: week1/tinajeong/pgm_42577_tinajeong.java
import java.util.Arrays;
import java.util.HashSet;

public class pgm_42577_tinajeong {
    public boolean solution(String[] phoneBook) {
        HashSet<String> phoneNumberSet = new HashSet<>(List.of(phoneBook));

        for (String phoneNumber : phoneNumberSet) {
            for (int i = 1; i <= phoneNumber.length(); i++) {
                String prefix = phoneNumber.substring(0, i);
                if (phoneNumberSet.contains(prefix) && !phoneNumber.equals(prefix)) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: week1/tinajeong/pgm_42577_tinajeong.py
def solution(phone_book):
    phone_number_set = set(phone_book)

    for phone_number in phone_number_set:
        for i in range(1, len(phone_number) + 1):
            prefix = phone_number[:i]
            if prefix in phone_number_set and phone_number != prefix:
                return False

    return True
